src/console_cmds.cpp

```diff
@@ -515,28 +515,36 @@ static bool ConKickOrBan(const char *argv, bool ban)
 DEF_CONSOLE_CMD(ConKick)
 {
 	if (argc == 0) {
-		IConsoleHelp("Kick a client from a network game. Usage: 'kick <ip | client-id>'");
+		IConsoleHelp("Kick a client from a network game. Usage: 'kick <ip | client-id> [<kick-reason>]'");
 		IConsoleHelp("For client-id's, see the command 'clients'");
 		return true;
 	}
 
-	if (argc != 2) return false;
+	if (argc != 2 && argc != 3) return false;
+
+	/* No reason supplied for kicking */
+	if (argc == 2) return ConKickOrBan(argv[1], false);
 
-	return ConKickOrBan(argv[1], false);
+	/* Reason for kicking supplied */
+	return ConKickOrBan(argv[1], false, argv[2]);
```

Comment:
I don't think anything "bad" would happen as a result, but if the reason given is longer than 255 chars, it'll be silently truncated. At the very least excess information is sent. Would be good to error message at this level rather than silently truncate

Reply:
I've added a check to prevent the admin from sending a truncated message. Instead it will complain that the kick message is longer than 254 characters.